src/openms/source/CHEMISTRY/AASequence.cpp

```diff
@@ -1034,6 +1035,7 @@ namespace OpenMS
     else if (specificity == ResidueModification::C_TERM)
     {
       ModificationsDB* mod_db = ModificationsDB::getInstance();
+      //TODO why are we allowing Protein Term here?
```

Comment:
I think because it can occur at the Protein Term but is not specific to that.

Reply:
Hmm ok we dont have knowledge about the protein here. Actually it is only used with "Any X-term", never with Protein N-term.
We should allow an additional optional parameter in the fromString method in case we know this peptide is at a protein terminus. Otherwise we may get Protein term mods in the middle of a protein, when no other mod matches.

Reply:
yeah I think this would be important